src/_pytest/tmpdir.py

```diff
@@ -2,21 +2,116 @@
 from __future__ import absolute_import, division, print_function
 
 import re
+from six.moves import reduce
```

Comment:
in py27+ you can consistently import this `from functools import reduce` -- would be a regression of https://github.com/pytest-dev/pytest/pull/3609